qiskit/pulse/instructions/instruction.py

```diff
@@ -204,7 +206,10 @@ def insert(self, start_time: int, schedule: ScheduleComponent,
         """
         if name is None:
             name = self.name
-        return Schedule(self, (start_time, schedule), name=name)
+        new_schedule = Schedule(name=name)
+        new_schedule.insert(0, self, inplace=True)
+        new_schedule.insert(start_time, schedule, inplace=True)
```

Comment:
As far as I could understand, this was the best way to have "`schedule` inserted within`self` at `start_time`"... is that right? (I double checked it's covered by having tests fail before by removing the second line)